sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,396 @@
+        it represents the absolute fraction of values that will be missing::
+            0.2
+
+        There will be ``0.2 * n_samples * n_features`` numbers of missing
+        values in the data.
+
+        Alternatively this refers to the probability of a value being dropped
+        after transform. The values are dropped (approximately) uniformly
+        across all labels and features. This type of missingness is referred
+        to as MCAR.
+
+        To vary the distribution across features or to prevent a feature from
+        having missing values, individual probabilities for each feature can
+        be specified as a 1D vector of shape ``(n_features,)``::
+            [0.2, 0.2, 0]
+
+        For the above example, the probability that a sample will have missing
+        value in feature 0 is 0.2. In other words, after calling
+        ``transform``, there are ``0.2 * n_samples`` randomly selected samples
+        with value missing in feature 0 and 1 but no samples with
+        missing values in feature 2.
+
+        If missingness is not MCAR, ``missing_distribution`` can be used
+        to specify the multinomial distribution of the newly dropped values
+        across labels (and if needed across features) as given below.
+
+        If ``missing_distribution`` is a dict of floats::
+            {1: 0.02, 2: 0.03, 3: 0.05}
+
+        The probability that a sample from class 1 will have a missing value
+        is 0.02.
+
+        The missing values are evenly spread across all the features.
+
+        In other words, there are ``int(0.02 / n_features * n_samples)``
+        randomly chosen samples of class 1 having missing values in each
+        feature.
+
+        If there are fewer than ``int(0.02 / n_features * n_samples)`` numbers
+        of samples in class 1, an error is raised.
+
+        Hence the total missing rate is ``0.02 + 0.03 + 0.05 = 0.1``.
+
+        If ``missing_distribution`` is a dict of vectors (and scalars)::
+
+            {0: 0.1,
+             3: [0.1, 0.15, 0.15]}
+
+        Note that the shape of the vector must be ``(n_features,)``
+
+        There are ``0.1 / n_features * n_samples`` randomly chosen samples,
+        for each feature, of class 1 having a missing value.
+
+        There are 0 samples of class 1 and 2 having missing value in any
+        feature.
+
+        And There are ``0.1 * n_samples`` randomly chosen samples having
+        missing value in feature 0, ``0.15 * n_samples`` randomly chosen
+        samples having missing value in feature 1 and feature 2, each.
+
+        A few notes:
+
+        Note that the samples are randomly chosen "*for each feature*".
+
+        The global ``missing_rate`` (fraction of missing values in the entire
+        dataset) is calculated as the sum of all the individual probabilities.
+
+        At the end of transform, the dataset will contain a total of
+        ``(X.shape[0] * X.shape[1]) * missing_rate`` numbers of missing values.
```

Comment:
Hmmm I can't think of a better way to shorten it without reducing clarity on the various formats :( Suggestions?